plugins/base/src/main/kotlin/transformers/documentables/DocumentableVisibilityFilterTransformer.kt

```diff
@@ -142,9 +148,37 @@ class DocumentableVisibilityFilterTransformer(val context: DokkaContext) : PreMe
 
         private fun filterProperties(
             properties: List<DProperty>,
-            additionalCondition: (DProperty, DokkaSourceSet) -> Boolean = ::alwaysTrue
-        ): Pair<Boolean, List<DProperty>> =
-            properties.transform(additionalCondition, ::hasVisibleAccessorsForPlatform) { original, filteredPlatforms ->
+            additionalCondition: (DProperty, DokkaSourceSet) -> Boolean = ::alwaysTrue,
+            additionalConditionAccessors: (DFunction, DokkaSourceSet) -> Boolean = ::alwaysTrue
+        ): Pair<Boolean, List<DProperty>> {
+
+            val modifier: (DProperty, Set<DokkaSourceSet>) -> Pair<Boolean, DProperty> =
+                { original, filteredPlatforms ->
+                    val setter = original.setter?.let { filterFunctions(listOf(it), additionalConditionAccessors) }
+                    val getter = original.getter?.let { filterFunctions(listOf(it), additionalConditionAccessors) }
+
+                    val modified = setter?.first == true || getter?.first == true
+
+                    Pair(modified,
+                        if (modified)
+                            with(original) {
+                                copy(
+                                    setter = setter?.second?.firstOrNull(),
+                                    getter = getter?.second?.firstOrNull()
+                                )
+                            }
+                        else original
+                    )
```

Comment:
Seems like `with(original) {}` does not make any difference, `original.copy` looks just fine and imho requires less brain power to understand :)

And I'd extract the result of if-else into a local variable and add indentation like below. Otherwise it looks a bit difficult to understand what is used where and how
```suggestion
                    val property =
                        if (modified)
                            original.copy(
                                setter = setter?.second?.firstOrNull(),
                                getter = getter?.second?.firstOrNull()
                            )
                        else original

                    modified to property
```